src/components/table-view.js

```diff
@@ -0,0 +1,166 @@
+
+    return (
+      <tr key={id} className={className} style={{ animationDelay }}>
+        <td className="ars-table-view-selection">
+          <Checker
+            checked={checked}
+            name={name}
+            id={id}
+            multiselect={multiselect}
+            onChange={onPicked}
+          />
+        </td>
+        <td className="ars-table-view-id">
+          {id}
+        </td>
+        <td className="ars-table-view-name">
+          {name}
+        </td>
+        <td>
+          {caption}
+        </td>
+        <td>
+          {attribution}
+        </td>
+        <td className="ars-table-view-preview">
+          <div className="ars-table-view-imagebox">
+            <img src={url} />
+          </div>
+        </td>
+      </tr>
+    )
+  }
+
+  sortBy = sortBy => {
+    this.setState({ sortBy })
+  }
+
+  render() {
+    const { items, multiselect, onKeyDown, onPicked, picked } = this.props
+    const { sortBy } = this.state
+
+    let rows = items.concat().sort(function(a, b) {
+      return `${a[sortBy]}` >= `${b[sortBy]}` ? 1 : -1
+    })
+
+    let ids = items.map(i => i.id)
+    let unselected = ids.filter(id => picked.indexOf(id) < 0)
+    let allPicked = unselected.length <= 0
+
+    return (
+      <div className="ars-table-view-wrapper" onKeyDown={onKeyDown}>
+        <table className="ars-table-view">
+          <thead>
+            <tr>
+              <th className="ars-table-view-selection">
+                <span className="ars-hidden">
+                  Use this column to select items
+                </span>
+
+                <Checker
+                  id={allPicked ? ids : unselected}
+                  name="all items"
+                  checked={allPicked}
+                  onChange={onPicked}
+                  disabled={!multiselect}
+                  multiselect
+                />
+              </th>
+              <Header field="id" active={sortBy === 'id'} onSort={this.sortBy}>
+                ID
+              </Header>
+              <Header
+                field="name"
+                active={sortBy === 'name'}
+                onSort={this.sortBy}
+              >
+                Name
+              </Header>
+              <Header
+                field="caption"
+                active={sortBy === 'caption'}
+                onSort={this.sortBy}
+              >
+                Caption
+              </Header>
+              <Header
+                field="attribution"
+                active={sortBy === 'attribution'}
+                onSort={this.sortBy}
+              >
+                Attribution
+              </Header>
+              <Header field="preview" active={sortBy === 'preview'}>
+                Preview
+              </Header>
+            </tr>
+          </thead>
+          <tbody>
+            {rows.map(this.renderRow, this)}
+          </tbody>
+        </table>
+      </div>
+    )
+  }
+}
+
+export default TableView
```

Comment:
Tables are really annoying... I want something concise, but each component needs to know the parent props. Seems like complexity abounds when I break up these components into smaller pieces. Something I'm going to muse on...